### ISSUE TITLE
Consolidate password restrictions

### ISSUE THREAD

**Issue by phet-steele** [phetsims/website-meteor#31]
The rules for making a password during registration are different than the rules for changing your password (after registration).

It appears that during registration my password:

- must be at least 8 characters
- can be blank

An example password that I've been using for registration that works is pressing the spacebar 8 times (8 empty characters).

After registration the rules change to at least:

- no length restriction
- CANNOT be blank

So in other words, the password of 8 spacebar presses is technically invalid at this point, yet something that's a single character, ex. "a", is valid.

@mattpen @oliver-phet the logic for both of these cases should be the same, and both have their pitfalls.

For phetsims/QA/issues/70.

**Comment by oliver-phet** [comment]
I would prefer these rules (always):
- must be at least 8 characters
- can be blank

**Comment by oliver-phet** [comment]
Sorry, meant "_can't_" be blank!

**Comment by mattpen** [comment]
It looks like the old form was trimming whitespace from passwords and the new form was not.  I've changed the new form so that it trims whitespace now.  For Steele's use case, an attempt to use a password with 8 spaces will now look like an attempt to use an empty string for the password and will fail.  This will be up on ox-dev soon.

**Comment by mattpen** [comment]
@phet-steele - This should be corrected now.  Please test and confirm.

**Comment by phet-steele** [comment]
> I would prefer these rules (always):
> 
> must be at least 8 characters

@mattpen this is still not true when changing your password (after registration). I made a password that was just "a".

**Comment by mattpen** [comment]
@phet-steele - This should be fixed on ox-dev.  Please test and confirm.

**Comment by phet-steele** [comment]
Works well.
